Adafruit_MQTT.cpp

```diff
@@ -267,7 +283,8 @@ uint16_t Adafruit_MQTT::readFullPacket(uint8_t *buffer, uint16_t maxsize,
   DEBUG_PRINT(F("Packet Length:\t"));
   DEBUG_PRINTLN(value);
 
-  if (value > (maxsize - (pbuff - buffer) - 1)) {
+  // maxsize is limited to 65536 by 16-bit unsigned
+  if (value > uint32_t(maxsize - (pbuff - buffer) - 1)) {
```

Comment:
`value` is `uint32_t` so it can handle the actual packet length. I made the cast explicit here in the conditional, but the key thing is that it's checked and for packets larger than 65536 (`maxsize` is `uint16_t`), you can't read past that size, and there is a debug message there.